flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/aggregations.scala

```diff
@@ -130,3 +142,63 @@ case class Avg(child: Expression) extends Aggregation {
     new SqlAvgAggFunction(AVG)
   }
 }
+
+case class UDAGGFunctionCall(
+    aggregateFunction: AggregateFunction[_, _],
+    args: Seq[Expression])
+  extends Aggregation {
+
+  override private[flink] def children: Seq[Expression] = args
+
+  // Override makeCopy method in TreeNode, to produce vargars properly
+  override def makeCopy(args: Array[AnyRef]): this.type = {
+    if (args.length < 1) {
+      throw new TableException("Invalid constructor params")
+    }
+    val agg = args.head.asInstanceOf[AggregateFunction[_, _]]
+    val arg = args.last.asInstanceOf[Seq[Expression]]
+    new UDAGGFunctionCall(agg, arg).asInstanceOf[this.type]
+  }
+
+  override def resultType: TypeInformation[_] = TypeExtractor.createTypeInfo(
```

Comment:
This is again very dangerous. Type extraction fails very often. We should provide a `getResultType` in `AggregateFunction` that the user can implement.